### ISSUE TITLE
 No statement label found error when generating SSP markdown

### ISSUE THREAD

**Issue by ride808** [IBM/compliance-trestle#1333]
## Describe the bug

When attempting to generate and assemble an ssp from component definitions I'm unable to consistently get statements to be added to my ssp markdown and subsequently the ssp model upon assembly.  IMO it seems to be related to whether an implemented control has sub statements (ex. statement ID like ra-5_smt.a  and ra-5_smt.b versus a statement ID like si-4.12_smt). Is there an issue in the catalog_writer for handing statement id's that do not have sub parts that is causing this error and subsequently my implementations not being captured by the ssp-generate and ssp-assemble into my model?

## To Reproduce

Create component:
```
{
  "component-definition": {
    "uuid": "346948d8-90c5-4bef-8101-f8954355cdf6",
    "metadata": {
      "title": "Alertmanager",
      "last-modified": "2023-03-15T13:44:21.728651+00:00",
      "version": "1.0.0",
      "oscal-version": "1.0.2"
    },
    "components": [
      {
        "uuid": "3289eb19-d740-4826-aded-aa362bca3934",
        "type": "software",
        "title": "AlertManager",
        "description": "",
        "control-implementations": [
          {
            "uuid": "3d78656b-0cce-43f7-99cf-ef7547f442bb",
            "source": "trestle://profiles/NIST_SP-800-53_rev5_Acme-baseline/profile.json",
            "description": "ACME Profile from catalog NIST Special Publication 800-53 Revision 5",
            "props": [
              {
                "name": "Rule_Id",
                "value": "xccdf_acme.alertmanager_rule_1000001",
                "remarks": "rule_set_0"
              },
              {
                "name": "Rule_Description",
                "value": "Alertmanager is installed in the system and is configured to alert based off of collected system events/metrics.",
                "remarks": "rule_set_0"
              }
            ],
            "implemented-requirements": [
              {
                "uuid": "814c7f73-d243-4eab-8f29-87eb26557bd9",
                "control-id": "si-4.12",
                "description": "Implementation: AlertManager is installed in the system to generate alerts based off of collected metrics",
                "props": [
                  {
                    "name": "Rule_Id",
                    "value": "xccdf_acme.alertmanager_rule_1000001"
                  },
                  {
                    "name": "implementation-status",
                    "value": "implemented"
                  }
                ],
                "statements": [
                  {
                    "statement-id": "si-4.12_smt",
                    "uuid": "d24902c0-a5e3-42c0-bc9d-b1a365193d15",
                    "description": "Implementation Statemetn: Alertmanager is installed in the system to generate alerts based off of collected metrics",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.alertmanager_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "implemented"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
```

Generate SSP with component:
```
trestle author ssp-generate --profile NIST_SP-800-53_rev5_Acme-baseline --compdefs "AlertManager" --output /tmp/myssp
```

Observe error:
```
trestle.core.catalog.catalog_interface:840 WARNING: No statement label found for statement id si-4.12_smt.  Defaulting to si-4.12_smt.
```

Excerpt below from my catalog showing si-4.12 and the statement-id I'm using.

```
              {
                "id": "si-4.12",
                "class": "SP800-53-enhancement",
                "title": "Automated Organization-generated Alerts",
                "params": [
                  {
                    "id": "si-04.12_odp.01",
                    "props": [
                      {
                        "name": "alt-identifier",
                        "value": "si-4.12_prm_1"
                      },
                      {
                        "name": "label",
                        "value": "SI-04(12)_ODP[01]",
                        "class": "sp800-53a"
                      }
                    ],
                    "label": "personnel or roles",
                    "guidelines": [
                      {
                        "prose": "personnel or roles to be alerted when indications of inappropriate or unusual activity with security or privacy implications occur is/are defined;"
                      }
                    ]
                  },
                  {
                    "id": "si-04.12_odp.02",
                    "props": [
                      {
                        "name": "alt-identifier",
                        "value": "si-4.12_prm_2"
                      },
                      {
                        "name": "label",
                        "value": "SI-04(12)_ODP[02]",
                        "class": "sp800-53a"
                      }
                    ],
                    "label": "automated mechanisms",
                    "guidelines": [
                      {
                        "prose": "automated mechanisms used to alert personnel or roles are defined;"
                      }
                    ]
                  },
                  {
                    "id": "si-04.12_odp.03",
                    "props": [
                      {
                        "name": "alt-identifier",
                        "value": "si-4.12_prm_3"
                      },
                      {
                        "name": "label",
                        "value": "SI-04(12)_ODP[03]",
                        "class": "sp800-53a"
                      }
                    ],
                    "label": "activities that trigger alerts",
                    "guidelines": [
                      {
                        "prose": "activities that trigger alerts to personnel or are defined;"
                      }
                    ]
                  }
                ],
                "props": [
                  {
                    "name": "label",
                    "value": "SI-4(12)"
                  },
                  {
                    "name": "label",
                    "value": "SI-04(12)",
                    "class": "sp800-53a"
                  },
                  {
                    "name": "sort-id",
                    "value": "si-04.12"
                  }
                ],
                "links": [
                  {
                    "href": "#si-4",
                    "rel": "required"
                  }
                ],
                "parts": [
                  {
                    "id": "si-4.12_smt",
                    "name": "statement",
                    "prose": "Alert {{ insert: param, si-04.12_odp.01 }} using {{ insert: param, si-04.12_odp.02 }} when the following indications of inappropriate or unusual activities with security or privacy implications occur: {{ insert: param, si-04.12_odp.03 }}."
                  },
                  {
                    "id": "si-4.12_gdn",
                    "name": "guidance",
                    "prose": "Organizational personnel on the system alert notification list include system administrators, mission or business owners, system owners, senior agency information security officer, senior agency official for privacy, system security officers, or privacy officers. Automated organization-generated alerts are the security alerts generated by organizations and transmitted using automated means. The sources for organization-generated alerts are focused on other entities such as suspicious activity reports and reports on potential insider threats. In contrast to alerts generated by the organization, alerts generated by the system in [SI-4(5)](#si-4.5) focus on information sources that are internal to the systems, such as audit records."
                  },
                  {
                    "id": "si-4.12_assessment-objective",
                    "name": "assessment-objective",
                    "prose": "{{ insert: param, si-04.12_odp.01 }} is/are alerted using {{ insert: param, si-04.12_odp.02 }} when {{ insert: param, si-04.12_odp.03 }} indicate inappropriate or unusual activities with security or privacy implications."
                  }
                ]
              },
```

The component below works fine in the ssp generate and the only difference I see is that the ra-5 control in my catalog has sub statements a-f.

```
{
  "component-definition": {
    "uuid": "136c8a3b-9fc3-4b06-99c6-d5a18469fc77",
    "metadata": {
      "title": "Aqua",
      "last-modified": "2023-03-20T18:56:26.441125+00:00",
      "version": "1.0.0",
      "oscal-version": "1.0.2"
    },
    "components": [
      {
        "uuid": "f8e32fb3-ef29-4bd6-8322-2356f775e361",
        "type": "software",
        "title": "Aqua",
        "description": "",
        "control-implementations": [
          {
            "uuid": "73ee0c0b-6b6e-4c24-9028-b7aa8ba8dbe0",
            "source": "trestle://profiles/NIST_SP-800-53_rev5_Acme-baseline/profile.json",
            "description": "ACME Profile from catalog NIST Special Publication 800-53 Revision 5",
            "props": [
              {
                "name": "Rule_Id",
                "value": "xccdf_acme.aqua_rule_1000001",
                "remarks": "rule_set_0"
              },
              {
                "name": "Rule_Description",
                "value": "The SBOM produced by every ACME environment contains container vulnerabilities found through Aqua scanning",
                "remarks": "rule_set_0"
              }
            ],
            "implemented-requirements": [
              {
                "uuid": "349195b4-e640-4e73-a2a8-d629fe131452",
                "control-id": "ra-5",
                "description": "Implementation: The SBOM produced by every ACME environment documents container vulnerabilities found through Aqua scanning.\n\nReasons for Partial Implementation Status:\n1) Aqua is currently only used to scan the ACME container and containers within the ACME SBOM\n2) Process for remediating vlunerabilities is planned",
                "props": [
                  {
                    "name": "Rule_Id",
                    "value": "xccdf_acme.aqua_rule_1000001"
                  },
                  {
                    "name": "implementation-status",
                    "value": "partial"
                  }
                ],
                "statements": [
                  {
                    "statement-id": "ra-5_smt.a",
                    "uuid": "0f63f9a0-2b53-496e-b411-27898e2fe93e",
                    "description": "Implementation: Aqua is used to monitor and scan for container vulnerabilities. Aqua is currently only used to scan the ACME container and containers in ACME SBOM.",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "partial"
                      }
                    ]
                  },
                  {
                    "statement-id": "ra-5_smt.b",
                    "uuid": "d4c1eed2-491a-416d-b71c-a64cc3720d91",
                    "description": "Implementation: Aqua is used to monitor and scan for container vulnerabilities and measure vulnerability impact within the system",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "partial"
                      }
                    ]
                  },
                  {
                    "statement-id": "ra-5_smt.c",
                    "uuid": "d163f91f-2417-45db-bf8e-f675bf424807",
                    "description": "Implementation: Vulnerabilities found by Aqua are reported within the ACME SBOM to allow for monitoring. Currently only deployed containers.",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "partial"
                      }
                    ]
                  },
                  {
                    "statement-id": "ra-5_smt.d",
                    "uuid": "f1b29f86-588b-4772-a454-7636a7d00434",
                    "description": "Not Currently implemented. Development of a process for remediatating vulnerabilities is planned",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "planned"
                      }
                    ]
                  },
                  {
                    "statement-id": "ra-5_smt.e",
                    "uuid": "f84aa631-103c-4364-89ab-10c026ba585f",
                    "description": "Implementation: Aqua vulnerabilities are captured in ACME release artifacts via an SBOM",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "implemented"
                      }
                    ]
                  },
                  {
                    "statement-id": "ra-5_smt.f",
                    "uuid": "727092da-8c08-4817-bac3-0389ada952bc",
                    "description": "Implementation: The Aqua vulnerability database is updated with vulnerabilities to be scanned",
                    "props": [
                      {
                        "name": "Rule_Id",
                        "value": "xccdf_acme.aqua_rule_1000001"
                      },
                      {
                        "name": "implementation-status",
                        "value": "implemented"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
```

## Expected behavior

Be able to generate ssp markdown with statements for controls that don't have sub statement parts (ex. a-e). 

## Environment

- OS: Linux RHEL8
- Python version: 3.8.10
- Installed packages: Trestle 2.0.0

**Comment by fsuits** [comment]
@ride808 sorry for the confusion here.  If a control statement has no sub parts then you can just define the rules at the top level of the implemented requirement and there is no need for an explicit statement in it.  I think that should fix it for you.  Just provide the description and the props for the rules in the imp_req and remove the statement from the imp_req.  Let us know if you have any issues after that change.

**Comment by ride808** [comment]
@fsuits I did run the jinja cmd using your branch and the implementations now populate as expected. Thanks for the quick fix!  Any idea of when you might drop a 2.0.1 patch release? Thanks!

**Comment by fsuits** [comment]
@ride808 great!  Thanks for confirming.  I will merge the change.  We have been aiming for  a release this week and it's good this change was caught.  I expect a new release within a few days.  Thanks!
